examples/replay.cc

```diff
@@ -14,21 +15,23 @@ class Replay : public sc2::ReplayObserver {
         sc2::ReplayObserver() {
     }
 
+    
+
+    //Called once when the game is started (or restarted)
     void OnGameStart() final {
-        const sc2::ObservationInterface* obs = Observation();
-        assert(obs->GetUnitTypeData().size() > 0);
-        count_units_built_.resize(obs->GetUnitTypeData().size());
-        std::fill(count_units_built_.begin(), count_units_built_.end(), 0);
     }
     
+    //Called when a unit is created by the player being observed
     void OnUnitCreated(const sc2::Unit& unit) final {
-        assert(uint32_t(unit.unit_type) < count_units_built_.size());
```

Comment:
Not sure why you deleted this. Now the logic in OnGameEnd won't output anything.